src/controller/CHIPDeviceController.cpp

```diff
@@ -243,11 +221,9 @@ CHIP_ERROR DeviceController::Shutdown()
 
     if (mFabricInfo != nullptr)
     {
-        // Shut down any ongoing CASE session activity we have.  Note that we
-        // own the mCASESessionManager, so shutting down everything on it is
-        // fine.  If we ever end up sharing the CASE session manager with other
-        // DeviceController instances we may need to be more targeted here.
-        mCASESessionManager->ReleaseAllSessions();
+        // Shut down any ongoing CASE session activity we have.  We're going to
+        // assume that all sessions for our fabric belong to us here.
+        mSystemState->CASESessionMgr()->ReleaseSessionsForFabric(GetCompressedFabricId());
```

Comment:
I actually have no idea how to address this issue.  Someone needs to not leak the sessions, and sessions are not really associated with a controller after this change....

But also, would these be multiple controllers all on the same fabric with the same node id?  Because if not, I think we already have assumptions about how there is at most one node identity per fabric for a Matter stack instance....

Reply:
The spec implies that a single node cannot have multiple identities on the same fabric (or maybe this is mandated explicitly; I forget).

But this doesn't really have much practical meaning because machines and nodes are not a thing that exist in a 1:1 way.  It should be entirely plausible to have multiple nodes just so long as they advertise on separate ports.  And at that point, it would be an implementation choice as to whether these need to have process isolation.  Certainly the sdk as written would require process isolation.  BUT we didn't need to implement things that way.

That's all pretty philosophical though...

Reply:
> just so long as they advertise on separate ports

True, but that's not what multiple CHIPDeviceController instances do.  They all listen on a single port.

Reply:
Chatted with @bzbarsky-apple about this offline, and we agreed that for 1.0, to not support multiple controllers / fabric in the same process in the SDK. There are a number of reasons why that's difficult to achieve, most fundamentally, to have each controller instance mapped to its own port when handling unsolicited messages, as well as other bits of changes needed around object lifetimes to make this all hang correctly.

He'll file an issue to have the REPL fixed (which does support this today, albeit, it doesn't rely on unsolicited message handling) and not support this modality.

Reply:
* Created https://github.com/project-chip/connectedhomeip/pull/16404 to prevent AddNewFabric of a fabric that collides with an existing fabric.
* Filed https://github.com/project-chip/connectedhomeip/issues/16405 on the python REPL
* Filed https://github.com/project-chip/connectedhomeip/issues/16406 to add checks to disallow this once hte REPL is fixed.